config/default/manager/manager.yaml

```diff
@@ -92,6 +92,13 @@ spec:
         - containerPort: 8000
           name: webhook-server
           protocol: TCP
+        readinessProbe:
+          exec:
+            command:
+            - cat
+            - /tmp/ready
+          initialDelaySeconds: 5
```

Comment:
why do we need any golang code for this hack? Running `cat` of an ever-existing file every 5 seconds feels silly.
Would it not be enough to have
```
   initialDelaySeconds: 20 
```
with a fat comment carrying the same explanation?

Reply:
We need to create the file once the server starts.

I don't think saving 3 `cat` runs is worth an extra line and comment.

We can replace `cat` with HTTP GET once controller-runtime implements support for it. I'm not aware of any option that would allow us to stop querying once it succeeds once. https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes

Reply:
I was speaking about something like
```
       readinessProbe:
          exec:
            command:
            - true
          initialDelaySeconds: 20
          periodSeconds: 99999
```
and dropping the golang changes altogether. But I'm splitting hairs here. The important thing is that you have found the races that @AlonaKaplan has spoken about.

Reply:
That would not help. True can succeed as soon as the pod starts. That would be the same as happens with default readiness.

The point here is, that we write the file only once we go through all the initialization and we start the webhook server. In other words, we are marked Ready only once we are able to respond to HTTPS requests for /mutate-virtualmachines.

Reply:
We still have races in the code itself. Unfortunately, the tests are more resilient than the code, so it is all hidden.

Reply:
The extra 20 seconds wait is on top of all the initialization (collecting existing MACs, getting certificates). Those 20 seconds are waiting until the `Start` hopefully really starts serving.

Reply:
gotcha.